### PR TITLE
[C-558] Add FeatureFlagOverrideModal

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description
* Add FeatureFlagOverrideModal (Opens with 'F' in dev mode) 
* Update the useFlag hook to respect the overrides

### Dragons

N/A

### How Has This Been Tested?

Manually tested

### How will this change be monitored?

N/A

### CODE REVIEW

File: packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const setOverrideSetting = (flag: FeatureFlags, val: OverrideSetting) => {
+  const flagKey = `${FEATURE_FLAG_OVERRIDE_KEY}:${flag}`
+  if (val === null) return localStorage.removeItem(flagKey)
+  localStorage.setItem(flagKey, val)
+}
+
+export const FeatureFlagOverrideModal = () => {
+  const [isOpen, setIsOpen] = useModalState('FeatureFlagOverride')
+
+  const [challengeRewardsUiSetting, setChallengeRewardsUiSetting] = useState(
```
Comment (sliptype) [inline packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx:39]: Is there any way we can map over the the FeatureFlag enum to create the initial state & info map? Would be ideal not to have to change another file when adding/removing flags.

Something like this maybe:

```ts
const [flagOverrideState, setFlagOverrideState] = useState(
  mapValues(FeatureFlags, (flag) => getOverrideSetting(flag))
)
```

File: packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      setter: setTippingSetting
+    }
+  }
+
+  const openModal = () => setIsOpen(true)
+  const closeModal = () => setIsOpen(false)
+
+  useEffect(() => {
+    window.addEventListener('keypress', (e: KeyboardEvent) => {
+      if (e.ctrlKey && e.key === 'f') isOpen ? closeModal() : openModal()
```
Comment (sliptype) [inline packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx:150]: Kinda worried about this being too accessible for the general public. I think we should do it like the discovery node selection popup does: https://github.com/AudiusProject/audius-client/blob/main/packages/web/src/components/discovery-node-selection/DiscoveryNodeSelection.tsx#L13

That requires the dev mode local storage item to be set for the hotkey to work

**Review by sliptype — APPROVED** [review]
Looks awesome! Had a couple ideas

File: packages/web/src/common/hooks/useFeatureFlag.ts [hunk 3]
```diff
@@ -20,17 +24,28 @@ import { StateWithRemoteConfig } from 'common/store/remote-config/slice'
...
 ) => (flag: FeatureFlags) => {
+  const overrideKey = `${FEATURE_FLAG_OVERRIDE_KEY}:${flag}`
   const configLoaded = useSelector((state: State) =>
     isRemoteConfigLoaded<State>(state)
   )
   const userIdFlag = flagCohortType[flag] === FeatureFlagCohortType.USER_ID
   const hasAccount = useSelector(getAccountUser)
   const shouldRecompute = userIdFlag ? hasAccount : true
+  const setOverride = (val: OverrideSetting) => {
+    if (val === null) return localStorage.removeItem(overrideKey)
```
Comment (raymondjacobson) [inline packages/web/src/common/hooks/useFeatureFlag.ts:35]: We don't want to depend directly on localstorage here in common right? Can we use the state context stuff? Or else this really shouldn't live in common.

File: packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx:150]: +1

File: packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx (hunk 1 above)
Comment (Kyle-Shanks) [reply packages/web/src/components/feature-flag-override-modal/FeatureFlagOverrideModal.tsx:39]: i was looking for a way to do this. I'll try this out and see if I can get it to work 👍

File: packages/web/src/common/hooks/useFeatureFlag.ts (hunk 3 above)
Comment (sliptype) [reply packages/web/src/common/hooks/useFeatureFlag.ts:35]: Oh good call. Yeah the `remoteConfigInstance` passed into here already has `getFeatureFlagSessionId` and `setFeatureFlagSessionId`. We could change those to simply be `getLocalStorageItem` and `setLocalStorageItem` and then expose/use those in this hook. @Kyle-Shanks lmk if you wanna sync on this so I can explain better

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-feature-flag-toggler

**Review by raymondjacobson — APPROVED** [review]
Screen Shot 2022-06-16 at 4 39 02 PM

Code LGTM! Default is a bit confusing, b/c it makes it a bit hard to tell off the bat what i do have enabled and what i don't, but im not sure if there's a better solution. Maybe automatically putting Enabled or Disabled based on what the default is and then a "Reset default" button at the bottom?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-feature-flag-toggler

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-feature-flag-toggler

**Review by sliptype — APPROVED** [review]
Nice! Excited to use this 🥳

**Pull request merged**